src/AssetCollection.php

```diff
@@ -139,32 +139,32 @@ public function remove($name)
      *
      * @return int
      */
-    public function count()
+    public function count(): int
     {
         return count($this->items);
     }
 
-    public function rewind()
+    public function rewind(): void
     {
         $this->index = 0;
     }
 
-    public function next()
+    public function next(): void
     {
         $this->index++;
     }
 
-    public function key()
+    public function key(): mixed
     {
         return $this->index;
     }
 
-    public function valid()
+    public function valid(): bool
     {
         return isset($this->items[$this->index]);
     }
 
-    public function current()
+    public function current(): mixed
```

Comment:
I don't think we can use `mixed` yet as it causes errors in PHP7.x. We might be able to use `#[\ReturnTypeWillChange]` attribute and no formal return type.

Reply:
Thanks for that. I've added the attributes and remove the mixed return type.